lalrpop/src/api/mod.rs

```diff
@@ -208,21 +212,24 @@ impl Configuration {
     /// Process all files in the current directory, which -- unless you
     /// have changed it -- is typically the root of the crate being compiled.
     pub fn process_current_dir(&self) -> Result<(), Box<dyn Error>> {
+        // If we can get a current dir, check to make sure session.in_dir either *wasn't* set, or
+        // wasn't set to that dir.  If we can't get a current dir, we'll error out in a moment
+        // anyways, and that's the bigger problem.
+        if let Ok(current_dir) = current_dir() {
+            if self.session.in_dir.is_some() && self.session.in_dir != Some(current_dir) {
+                eprintln!("Error: \"process_current_dir()\" contradicts previously set in_dir");
```

Comment:
Is it accurate that this can be reached from `process_root()` as well? If so, lets not used the function name explicitly(or I guess add checking to process_root as well but that would duplicate the logic)

Reply:
Hmm, actually I guess it's reachable but will never occur because `process_root` creates its own Configuration object and an `in_dir` isn't set

Reply:
Looking over everything, I do think this is the only place the message is reachable, as you said above.